tests/test_workspaces.rs

```diff
@@ -0,0 +1,191 @@
+
+    test_env.jj_cmd_success(&main_path, &["workspace", "add", "../secondary"]);
+
+    let stdout = test_env.jj_cmd_success(&main_path, &["log", "-T", r#"commit_id " " checkouts"#]);
+    insta::assert_snapshot!(stdout, @r###"
+    o 6bafff1a880f313aebb6d357c79b7aa4befa0af8 secondary@
+    | @ c8f1217f93a0bc570a8bbfe055980f27062339ef default@
+    |/  
+    o 5af56dcc2cc27bb234e5574b5a3ebc5f22081462 
+    o 0000000000000000000000000000000000000000 
+    "###);
+
+    // Make changes in both working copies
+    std::fs::write(main_path.join("file"), "changed in main\n").unwrap();
+    std::fs::write(secondary_path.join("file"), "changed in second\n").unwrap();
+    // Squash the changes from the main workspace in the initial commit (before
+    // running any command in the secondary workspace
+    let stdout = test_env.jj_cmd_success(&main_path, &["squash"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Rebased 1 descendant commits
+    Working copy now at: 86bef7fee095 
+    "###);
+
+    // The secondary workspace's checkout was updated
+    let stdout = test_env.jj_cmd_success(&main_path, &["log", "-T", r#"commit_id " " checkouts"#]);
+    insta::assert_snapshot!(stdout, @r###"
+    @ 86bef7fee095bb5626d853c222764fc7c9fb88ac default@
+    | o 8d8269a323a01a287236c4fd5f64dc9737febb5b secondary@
+    |/  
+    o 52601f748bf6cb00ad5389922f530f20a7ecffaa 
+    o 0000000000000000000000000000000000000000 
+    "###);
+    let stdout = test_env.jj_cmd_success(
+        &secondary_path,
+        &["log", "-T", r#"commit_id " " checkouts"#],
+    );
+    // It was detected that the working copy is now stale
+    // TODO: Since there was an uncommitted change in the working copy, it should
+    // have been committed first (causing divergence)
+    assert!(stdout.starts_with("The working copy is stale"));
+    insta::assert_snapshot!(stdout.lines().skip(1).join("\n"), @r###"
+    o 86bef7fee095bb5626d853c222764fc7c9fb88ac default@
+    | @ 8d8269a323a01a287236c4fd5f64dc9737febb5b secondary@
+    |/  
+    o 52601f748bf6cb00ad5389922f530f20a7ecffaa 
+    o 0000000000000000000000000000000000000000 
+    "###);
+}
+
+/// Test forgetting workspaces
+#[test]
+fn test_workspaces_forget() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "--git", "main"]);
+    let main_path = test_env.env_root().join("main");
+
+    std::fs::write(main_path.join("file"), "contents").unwrap();
+    test_env.jj_cmd_success(&main_path, &["close", "-m", "initial"]);
+
+    test_env.jj_cmd_success(&main_path, &["workspace", "add", "../secondary"]);
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "forget"]);
+    insta::assert_snapshot!(stdout, @"");
+
+    // When listing workspaces, only the secondary workspace shows up
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @"secondary: 39a6d6c6f295 
+");
```

Comment:
If we have a solution for the "); in the first column above, don't forget to apply it in this function.